balboa-agent/src/main/java/com/socrata/balboa/agent/MetricConsumer.java

```diff
@@ -147,7 +157,6 @@ public Record(String entityId, String name, Number value, long timestamp, Metric
             while (true) {
                 Map<String, String> record = grovel(stream);
                 if (record == null) {
-                    log.error("Unable to process records from {}", filePath);
                     break;
```

Comment:
If you throw an exception for all the error conditions in grovel, and return null for all the EOF conditions, your while loop becomes:

Map<String, String> record;
while ((record = grovel(stream)) != null) {
...
}